? hello.num
? cpukit/j.diff
? cpukit/log
? testsuites/sptests/hold
Index: c/src/make/leaf.cfg
===================================================================
RCS file: /usr1/CVS/rtems/c/src/make/leaf.cfg,v
retrieving revision 1.5
diff -u -r1.5 leaf.cfg
--- c/src/make/leaf.cfg	9 May 2007 18:43:18 -0000	1.5
+++ c/src/make/leaf.cfg	4 Feb 2008 14:35:53 -0000
@@ -41,8 +41,9 @@
 # Create a RTEMS executable based on MANAGERS which was set in
 #  app's Makefile
 
-MANAGERS_NOT_WANTED=$(filter-out $(MANAGERS) \
-	$(MANAGERS_REQUIRED), $(MANAGER_LIST))
+## XXX Joel make conditional
+## MANAGERS_NOT_WANTED=$(filter-out $(MANAGERS) \
+## 	$(MANAGERS_REQUIRED), $(MANAGER_LIST))
 
 # Start file must be one of
 #    $(PROJECT_RELEASE)/lib/start$(LIB_VARIANT).o
Index: c/src/make/target.cfg.in
===================================================================
RCS file: /usr1/CVS/rtems/c/src/make/target.cfg.in,v
retrieving revision 1.7
diff -u -r1.7 target.cfg.in
--- c/src/make/target.cfg.in	4 Aug 2007 05:45:16 -0000	1.7
+++ c/src/make/target.cfg.in	4 Feb 2008 14:35:53 -0000
@@ -55,6 +55,7 @@
 # Create a RTEMS executable based on MANAGERS which was set in
 #  app's Makefile
 
-MANAGERS_NOT_WANTED=$(filter-out $(MANAGERS) \
-	$(MANAGERS_REQUIRED), $(MANAGER_LIST))
+## XXX Joel make conditional
+## MANAGERS_NOT_WANTED=$(filter-out $(MANAGERS) \
+## 	$(MANAGERS_REQUIRED), $(MANAGER_LIST))
 
Index: cpukit/libcsupport/preinstall.am
===================================================================
RCS file: /usr1/CVS/rtems/cpukit/libcsupport/preinstall.am,v
retrieving revision 1.11
diff -u -r1.11 preinstall.am
--- cpukit/libcsupport/preinstall.am	31 Jan 2008 16:17:34 -0000	1.11
+++ cpukit/libcsupport/preinstall.am	4 Feb 2008 14:35:54 -0000
@@ -8,10 +8,24 @@
 PREINSTALL_DIRS =
 DISTCLEANFILES = $(PREINSTALL_DIRS)
 
+all-local: $(TMPINSTALL_FILES)
+
+TMPINSTALL_FILES =
+CLEANFILES = $(TMPINSTALL_FILES)
+
 all-am: $(PREINSTALL_FILES)
 
 PREINSTALL_FILES =
-CLEANFILES = $(PREINSTALL_FILES)
+CLEANFILES += $(PREINSTALL_FILES)
+
+$(PROJECT_LIB)/$(dirstamp):
+	@$(MKDIR_P) $(PROJECT_LIB)
+	@: > $(PROJECT_LIB)/$(dirstamp)
+PREINSTALL_DIRS += $(PROJECT_LIB)/$(dirstamp)
+
+$(PROJECT_LIB)/libcsupport.a: libcsupport.a $(PROJECT_LIB)/$(dirstamp)
+	$(INSTALL_DATA) $< $(PROJECT_LIB)/libcsupport.a
+TMPINSTALL_FILES += $(PROJECT_LIB)/libcsupport.a
 
 $(PROJECT_INCLUDE)/rtems/$(dirstamp):
 	@$(MKDIR_P) $(PROJECT_INCLUDE)/rtems
Index: cpukit/posix/preinstall.am
===================================================================
RCS file: /usr1/CVS/rtems/cpukit/posix/preinstall.am,v
retrieving revision 1.22
diff -u -r1.22 preinstall.am
--- cpukit/posix/preinstall.am	31 Jan 2008 16:17:34 -0000	1.22
+++ cpukit/posix/preinstall.am	4 Feb 2008 14:35:54 -0000
@@ -8,10 +8,20 @@
 PREINSTALL_DIRS =
 DISTCLEANFILES = $(PREINSTALL_DIRS)
 
+all-local: $(TMPINSTALL_FILES)
+
+TMPINSTALL_FILES =
+CLEANFILES = $(TMPINSTALL_FILES)
+
 all-am: $(PREINSTALL_FILES)
 
 PREINSTALL_FILES =
-CLEANFILES = $(PREINSTALL_FILES)
+CLEANFILES += $(PREINSTALL_FILES)
+
+$(PROJECT_LIB)/$(dirstamp):
+	@$(MKDIR_P) $(PROJECT_LIB)
+	@: > $(PROJECT_LIB)/$(dirstamp)
+PREINSTALL_DIRS += $(PROJECT_LIB)/$(dirstamp)
 
 $(PROJECT_INCLUDE)/$(dirstamp):
 	@$(MKDIR_P) $(PROJECT_INCLUDE)
@@ -19,6 +29,10 @@
 PREINSTALL_DIRS += $(PROJECT_INCLUDE)/$(dirstamp)
 
 if LIBPOSIX
+$(PROJECT_LIB)/libposix.a: libposix.a $(PROJECT_LIB)/$(dirstamp)
+	$(INSTALL_DATA) $< $(PROJECT_LIB)/libposix.a
+TMPINSTALL_FILES += $(PROJECT_LIB)/libposix.a
+
 if HAS_PTHREADS
 $(PROJECT_INCLUDE)/sched.h: include/sched.h $(PROJECT_INCLUDE)/$(dirstamp)
 	$(INSTALL_DATA) $< $(PROJECT_INCLUDE)/sched.h
Index: cpukit/rtems/Makefile.am
===================================================================
RCS file: /usr1/CVS/rtems/cpukit/rtems/Makefile.am,v
retrieving revision 1.42
diff -u -r1.42 Makefile.am
--- cpukit/rtems/Makefile.am	29 Jan 2008 21:52:20 -0000	1.42
+++ cpukit/rtems/Makefile.am	4 Feb 2008 14:35:54 -0000
@@ -7,6 +7,7 @@
 
 AM_CPPFLAGS += -D__RTEMS_INSIDE__
 
+project_lib_LIBRARIES = librtems.a
 noinst_LIBRARIES = librtems.a
 librtems_a_CPPFLAGS = $(AM_CPPFLAGS)
 
Index: cpukit/rtems/preinstall.am
===================================================================
RCS file: /usr1/CVS/rtems/cpukit/rtems/preinstall.am,v
retrieving revision 1.11
diff -u -r1.11 preinstall.am
--- cpukit/rtems/preinstall.am	29 Jan 2008 21:52:20 -0000	1.11
+++ cpukit/rtems/preinstall.am	4 Feb 2008 14:35:54 -0000
@@ -8,16 +8,30 @@
 PREINSTALL_DIRS =
 DISTCLEANFILES = $(PREINSTALL_DIRS)
 
+all-local: $(TMPINSTALL_FILES)
+
+TMPINSTALL_FILES =
+CLEANFILES = $(TMPINSTALL_FILES)
+
 all-am: $(PREINSTALL_FILES)
 
 PREINSTALL_FILES =
-CLEANFILES = $(PREINSTALL_FILES)
+CLEANFILES += $(PREINSTALL_FILES)
+
+$(PROJECT_LIB)/$(dirstamp):
+	@$(MKDIR_P) $(PROJECT_LIB)
+	@: > $(PROJECT_LIB)/$(dirstamp)
+PREINSTALL_DIRS += $(PROJECT_LIB)/$(dirstamp)
 
 $(PROJECT_INCLUDE)/$(dirstamp):
 	@$(MKDIR_P) $(PROJECT_INCLUDE)
 	@: > $(PROJECT_INCLUDE)/$(dirstamp)
 PREINSTALL_DIRS += $(PROJECT_INCLUDE)/$(dirstamp)
 
+$(PROJECT_LIB)/librtems.a: librtems.a $(PROJECT_LIB)/$(dirstamp)
+	$(INSTALL_DATA) $< $(PROJECT_LIB)/librtems.a
+TMPINSTALL_FILES += $(PROJECT_LIB)/librtems.a
+
 $(PROJECT_INCLUDE)/rtems.h: include/rtems.h $(PROJECT_INCLUDE)/$(dirstamp)
 	$(INSTALL_DATA) $< $(PROJECT_INCLUDE)/rtems.h
 PREINSTALL_FILES += $(PROJECT_INCLUDE)/rtems.h
Index: cpukit/sapi/Makefile.am
===================================================================
RCS file: /usr1/CVS/rtems/cpukit/sapi/Makefile.am,v
retrieving revision 1.33
diff -u -r1.33 Makefile.am
--- cpukit/sapi/Makefile.am	13 Dec 2007 17:00:38 -0000	1.33
+++ cpukit/sapi/Makefile.am	4 Feb 2008 14:35:54 -0000
@@ -19,6 +19,7 @@
 ## src
 AM_CPPFLAGS += -D__RTEMS_INSIDE__
 
+project_lib_LIBRARIES = libsapi.a
 noinst_LIBRARIES = libsapi.a
 libsapi_a_SOURCES = src/debug.c src/extension.c src/extensioncreate.c \
     src/extensiondelete.c src/extensionident.c src/fatal.c src/exinit.c \
Index: cpukit/sapi/preinstall.am
===================================================================
RCS file: /usr1/CVS/rtems/cpukit/sapi/preinstall.am,v
retrieving revision 1.9
diff -u -r1.9 preinstall.am
--- cpukit/sapi/preinstall.am	13 Dec 2007 17:00:38 -0000	1.9
+++ cpukit/sapi/preinstall.am	4 Feb 2008 14:35:54 -0000
@@ -8,10 +8,20 @@
 PREINSTALL_DIRS =
 DISTCLEANFILES = $(PREINSTALL_DIRS)
 
+all-local: $(TMPINSTALL_FILES)
+
+TMPINSTALL_FILES =
+CLEANFILES = $(TMPINSTALL_FILES)
+
 all-am: $(PREINSTALL_FILES)
 
 PREINSTALL_FILES =
-CLEANFILES = $(PREINSTALL_FILES)
+CLEANFILES += $(PREINSTALL_FILES)
+
+$(PROJECT_LIB)/$(dirstamp):
+	@$(MKDIR_P) $(PROJECT_LIB)
+	@: > $(PROJECT_LIB)/$(dirstamp)
+PREINSTALL_DIRS += $(PROJECT_LIB)/$(dirstamp)
 
 $(PROJECT_INCLUDE)/rtems/$(dirstamp):
 	@$(MKDIR_P) $(PROJECT_INCLUDE)/rtems
@@ -54,3 +64,7 @@
 	$(INSTALL_DATA) $< $(PROJECT_INCLUDE)/rtems/extension.inl
 PREINSTALL_FILES += $(PROJECT_INCLUDE)/rtems/extension.inl
 
+$(PROJECT_LIB)/libsapi.a: libsapi.a $(PROJECT_LIB)/$(dirstamp)
+	$(INSTALL_DATA) $< $(PROJECT_LIB)/libsapi.a
+TMPINSTALL_FILES += $(PROJECT_LIB)/libsapi.a
+
Index: cpukit/score/Makefile.am
===================================================================
RCS file: /usr1/CVS/rtems/cpukit/score/Makefile.am,v
retrieving revision 1.58
diff -u -r1.58 Makefile.am
--- cpukit/score/Makefile.am	30 Jan 2008 14:56:57 -0000	1.58
+++ cpukit/score/Makefile.am	4 Feb 2008 14:35:54 -0000
@@ -65,6 +65,7 @@
 
 AM_CPPFLAGS += -D__RTEMS_INSIDE__
 
+project_lib_LIBRARIES = libscore.a
 noinst_LIBRARIES = libscore.a
 libscore_a_SOURCES =
 libscore_a_CPPFLAGS = $(AM_CPPFLAGS)
Index: cpukit/score/preinstall.am
===================================================================
RCS file: /usr1/CVS/rtems/cpukit/score/preinstall.am,v
retrieving revision 1.16
diff -u -r1.16 preinstall.am
--- cpukit/score/preinstall.am	22 Jan 2008 18:28:53 -0000	1.16
+++ cpukit/score/preinstall.am	4 Feb 2008 14:35:55 -0000
@@ -8,10 +8,20 @@
 PREINSTALL_DIRS =
 DISTCLEANFILES += $(PREINSTALL_DIRS)
 
+all-local: $(TMPINSTALL_FILES)
+
+TMPINSTALL_FILES =
+CLEANFILES = $(TMPINSTALL_FILES)
+
 all-am: $(PREINSTALL_FILES)
 
 PREINSTALL_FILES =
-CLEANFILES = $(PREINSTALL_FILES)
+CLEANFILES += $(PREINSTALL_FILES)
+
+$(PROJECT_LIB)/$(dirstamp):
+	@$(MKDIR_P) $(PROJECT_LIB)
+	@: > $(PROJECT_LIB)/$(dirstamp)
+PREINSTALL_DIRS += $(PROJECT_LIB)/$(dirstamp)
 
 $(PROJECT_INCLUDE)/rtems/$(dirstamp):
 	@$(MKDIR_P) $(PROJECT_INCLUDE)/rtems
@@ -277,3 +287,7 @@
 	$(INSTALL_DATA) $< $(PROJECT_INCLUDE)/rtems/score/threadmp.inl
 PREINSTALL_FILES += $(PROJECT_INCLUDE)/rtems/score/threadmp.inl
 endif
+$(PROJECT_LIB)/libscore.a: libscore.a $(PROJECT_LIB)/$(dirstamp)
+	$(INSTALL_DATA) $< $(PROJECT_LIB)/libscore.a
+TMPINSTALL_FILES += $(PROJECT_LIB)/libscore.a
+
Index: make/compilers/gcc-no_bsp.cfg
===================================================================
RCS file: /usr1/CVS/rtems/make/compilers/gcc-no_bsp.cfg,v
retrieving revision 1.12
diff -u -r1.12 gcc-no_bsp.cfg
--- make/compilers/gcc-no_bsp.cfg	9 May 2007 18:53:32 -0000	1.12
+++ make/compilers/gcc-no_bsp.cfg	4 Feb 2008 14:35:56 -0000
@@ -231,8 +231,9 @@
 
 # Create a RTEMS executable based on MANAGERS which was set in app's Makefile
 
-MANAGERS_NOT_WANTED=$(filter-out $(MANAGERS), $(MANAGER_LIST))
-MANAGERS_NOT_WANTED:=$(filter-out $(MANAGERS_REQUIRED), $(MANAGERS_NOT_WANTED))
+## XXX JOEL make conditional
+## MANAGERS_NOT_WANTED=$(filter-out $(MANAGERS), $(MANAGER_LIST))
+## MANAGERS_NOT_WANTED:=$(filter-out $(MANAGERS_REQUIRED), $(MANAGERS_NOT_WANTED))
 
 # Start file must be one of
 # Note:  Normally RTEMS provides a start file...
Index: make/custom/erc32.cfg
===================================================================
RCS file: /usr1/CVS/rtems/make/custom/erc32.cfg,v
retrieving revision 1.34
diff -u -r1.34 erc32.cfg
--- make/custom/erc32.cfg	9 May 2007 18:46:13 -0000	1.34
+++ make/custom/erc32.cfg	4 Feb 2008 14:35:56 -0000
@@ -18,13 +18,15 @@
 CPU_CFLAGS = -mcpu=cypress
 
 # optimize flag: typically -O2
-CFLAGS_OPTIMIZE_V = -O2 -g
+CFLAGS_OPTIMIZE_V = -Os -g
 
 # The following are definitions of make-exe which will work using ld as
 # is currently required.
 
 define make-exe
 	$(LINK.c) $(AM_CFLAGS) $(AM_LDFLAGS) -o $(basename $@).exe \
+          -Wl,--undefined=rtems_task_create -Wl,--undefined=pthread_create \
+	  -Wl,--whole-archive -lrtems -lsapi -lscore -lposix -lcsupport -Wl,--no-whole-archive \
 	    $(LINK_OBJS) $(LINK_LIBS) -Wl,-Map,$(basename $@).map
 	$(NM) -g -n $(basename $@).exe > $(basename $@).num
 	$(SIZE) $(basename $@).exe
